lib/common/scores.c

```diff
@@ -18,51 +18,86 @@ int pcmk__score_green = 0;
 int pcmk__score_yellow = 0;
 
 /*!
- * \brief Get the integer value of a score string
+ * \brief Parse an integer score from a string
  *
- * Given a string representation of a score, return the integer equivalent.
- * This accepts infinity strings as well as red, yellow, and green, and
- * bounds the result to +/-INFINITY.
+ * Parse an integer score from a string. This accepts infinity strings as well
+ * as red, yellow, and green, and bounds the result to +/-INFINITY.
  *
- * \param[in] score  Score as string
+ * \param[in]  score_s   Score as string
+ * \param[out] score     Where to store integer value corresponding to \p score
+ *                       (may be NULL just to validate string)
+ * \param[in]  default   Value to use if \p score_s is NULL or invalid
  *
- * \return Integer value corresponding to \p score
+ * \return Standard Pacemaker return code
  */
 int
-char2score(const char *score)
+pcmk_parse_score(const char *score_s, int *score, int default_score)
 {
-    if (score == NULL) {
-        return 0;
+    int rc = pcmk_rc_ok;
+    int local_score = default_score;
```

Comment:
If `default_score` is out of bounds, then the end result is out of bounds in the case of `NULL` score or parsing error. We could assert that it's in bounds at the beginning, or force bounding later.

Reply:
Good catch